### DISCUSSION TITLE
Changing keyboard layout

### DISCUSSION THREAD

**Discussion by willabytes** [winapps-org/winapps#370] [Support] [answered]
Hello! I recently set up WinApps and it works really good... except for one thing. It uses the wrong keyboard layout, and everything I do to try to change it fails. At first I thought I'd just fire up the entire Windows environment and change it in Windows, but that _only_ changes it in the Windows environment itself. Then I read up a bit on FreeRDP, and tried to add the RDP flag to winapps.conf that in theory should've overridden the default US one, which was "/kbd:0x0000041D", but after that I couldn't initiate sessions at all. When I started Word for example, it just wouldn't launch. I also tried "/kbd:Swedish" and "/kbd:sv-SE", but no luck there either. How can I change the keyboard layout for WinApps?

Host:
OS: Arch
DE: Hyprland (which is on Wayland if that has anything to do with it)
FreeRDP version: 3.10.3 

Thanks in advance! :)

**Comment by oskardotglobal** [comment]
Did you properly load the registry files? Those should enforce the keyboard layout set in Windows

**Comment by Lumipyry** [comment, marked as answer]
I had the same problem - I'm using finnish keyboard (probably the same as swedish keyboard) - and yes - the solution was to fire up Windows and install finnish language pack - and make finnish keyboard default.
This applies at least to Podman and libvirt.

Another solution found in [181](https://github.com/winapps-org/winapps/discussions/181):
_"Keyboard layout not respected by FreeRDP. Fixed with [HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Keyboard Layout] add a new DWORD IgnoreRemoteKeyboardLayout with value 1"_ (inside Windows).

**Reply by willabytes** [reply]
Ohhh thanks! I think I fixed it. :)
